pkg/integration/prometheus/relabel_analyzer.go

```diff
@@ -0,0 +1,85 @@
+package prometheus
+
+import (
+	"fmt"
+
+	"github.com/k8sgpt-ai/k8sgpt/pkg/common"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
+	discoverykube "github.com/prometheus/prometheus/discovery/kubernetes"
+	"gopkg.in/yaml.v2"
+)
+
+type RelabelAnalyzer struct {
+}
+
+func (r *RelabelAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
+	ctx := a.Context
+	client := a.Client.GetClient()
+	namespace := a.Namespace
+	kind := ConfigRelabel
+
+	podConfigs, err := findPrometheusPodConfigs(ctx, client, namespace)
```

Comment:
Since both analyzers require the same codebase (findPrometheusPodConfigs, parsePromConfig, ...), shouldn't we find a way to retrieve this information once and pass it to the analyzers?

Reply:
Totally - I wasn't sure how to do that....

We could put a call to `findPrometheusPodConfigs` in the integration's `Deploy()` method and we can initiate each analyzer with that state. However, then IIUC if one runs

```
k8sgpt integration activate prometheus --no-install
```

the state won't be [1] initialized...

I'm open to suggestions 😄 

[1] https://github.com/k8sgpt-ai/k8sgpt/blob/v0.3.25/pkg/integration/integration.go#L86-L90

Reply:
Actually - thinking about this more, it may be beneficial to rerun this with every analyze call, in case the state of the cluster changes. Called that out here [1].

[1] https://github.com/k8sgpt-ai/k8sgpt/pull/855/files#diff-666d125bbcc3df99f3456b63ade46fc4e7c0d2e27ab325c709c0e792b9143524R44